specs/tables.md

```diff
@@ -0,0 +1,143 @@
+| $counter | false     | CallContext                | $callID  | LastCalleeReturnDataLength | $value      | 0           | 0           |
+|          |           |                            |          |                            |             |             |             |
+|          |           | *CallContext state*        |          | *CallContextFieldTag* (rw) |             |             |             |
+| $counter | $isWrite  | CallContext                | $callID  | IsRoot                     | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | IsCreate                   | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | CodeSource                 | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | ProgramCounter             | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | StackPointer               | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | GasLeft                    | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | MemorySize                 | $value      | 0           | 0           |
+| $counter | $isWrite  | CallContext                | $callID  | StateWriteCounter          | $value      | 0           | 0           |
+|          |           |                            |          |                            |             |             |             |
+| $counter | $isWrite  | Stack                      | $callID  | $stackPointer              | $value      | 0           | 0           |
+| $counter | $isWrite  | Memory                     | $callID  | $address                   | $value      | 0           | 0           |
+| $counter | $isWrite  | AccountStorage             | $address | $key                       | $value      | $valuePrev  | 0           |
+
+## `bytecode_table`
+
+Proved by the bytecode circuit.
+
+> - **isCode**: A boolean value to specify if the value is executable opcode or
+>   the data portion of PUSH\* operations.
+
+| 0 codeHash | 1 byteIndex | 2 value | 3 isCode |
+| ---        | ---         | ---     | ---      |
+| $codeHash  | $byteIndex  | $value  | $isCode  |
+
+## `block_table`
+
+Proved by the block circuit.
+
+| 0 Tag                  | 1      | 2 value |
+| ---                    | ---    | ---     |
+| *BlockContextFieldTag* |        |         |
+| Coinbase               | 0      | $value  |
+| GasLimit               | 0      | $value  |
+| BlockNumber            | 0      | $value  |
+| Time                   | 0      | $value  |
+| Difficulty             | 0      | $value  |
+| BaseFee                | 0      | $value  |
+| BlockHash              | 0..256 | $value  |
```

Comment:
This is currently implemented as the actual block number. For example, if current block has number `300`, then the number here will be `44..300`, like:

https://github.com/appliedzkp/zkevm-specs/blob/ffcc9295442bccd7bf23646ee9eb2565aef9ff82/src/zkevm_specs/evm/typing.py#L60-L63

But after more thought on this, I think we should keep it simpler for verifier to have as `0..256`, since the constraints of `BLOCKHASH` would be almost same (differs on whether to subtract `block.number` or not).

Will try to update it in further PRs.